### PR TITLE
Add script fired events and add script inspector devtool

### PR DESCRIPTION
**Pull request by Trevor159**
Relies on !96

Picture of the dev tool 

Swing sucks and I suck at swing.

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/events/ScriptPreFired.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+{
+	/**
+	 * The script id of the invoked script
+	 */
+	private final int scriptId;
+
+	/**
+	 * The source of the script invoke, this will be null unless
+	 */
+	private Widget source = null;
```
Comment (abextm) [inline runelite-api/src/main/java/net/runelite/api/events/ScriptPreFired.java]: this should probably hold the whole scriptevent

File: runelite-client/src/main/java/net/runelite/client/plugins/devtools/ScriptInspector.java [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+import net.runelite.client.ui.ClientUI;
+import net.runelite.client.ui.ColorScheme;
+import net.runelite.client.ui.DynamicGridLayout;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.util.Text;
+
+@Slf4j
+public class ScriptInspector extends JFrame
+{
+	private final static String DEFAULT_BLACKLIST = "3174,1004";
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/devtools/ScriptInspector.java]: Name these in ScriptID so they're not magic numbers.

File: runelite-client/src/main/java/net/runelite/client/plugins/devtools/ScriptInspector.java [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+	private final static int MAX_LOG_ENTRIES = 10000;
+
+	private final Client client;
+	private final EventBus eventBus;
+	private final ConfigManager configManager;
+
+	private final JPanel tracker = new JPanel();
+	private ScriptTreeNode currentNode;
+	private int lastTick = 0;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/devtools/ScriptInspector.java]: 0 is the initial value for int fields.
```suggestion
	private int lastTick;
```

File: runelite-client/src/main/java/net/runelite/client/plugins/devtools/ScriptInspector.java (hunk 2 above)
Comment (abextm) [reply runelite-client/src/main/java/net/runelite/client/plugins/devtools/ScriptInspector.java]: imo its better not in scriptid so I don't have to see them when they change, it doesn't matter for this use case.

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/devtools/ScriptInspector.java]: A comment explaining the IDs here would be sufficient then. It's just awkward to leave these as magic numbers.

**Pull request merged**